Code-4-Community/frontend-scaffold
# Implement Refresh and State Persistence

## Why

Previously, when you refresh the page you'd lose your login and any data. 



## This PR

First, this PR persists the Redux store to local storage, which enables the user to refresh the page and still retain their page data and access tokens. 

This PR additionally adds refresh functionality. Upon receiving a failed response due to an expired access token, we call the backend's refresh method to receive a new access token, then retry the failed request. 

Also I added change password to test the above functionality, needed a route for that. 



## Verification Steps

Refreshed the browser to observe that the Redux store was properly persisted. 

Loaded the application with an expired access token and a valid refresh token. Made a protected API request (change password), which failed, then a new access token was properly fetched and the request succeeded.
